### PR TITLE
Add TagTable migration app

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus have hard failures on this PR

File: src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/MigratorCompatibilitySpec.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        
+        protected override async Task InitializeTestAsync()
+        {
+            await base.InitializeTestAsync();
+
+            try
+            {
+                var config = _config.WithFallback(Config());
+                var migrator = new TagTableMigrator(config);
+                migrator.Migrate(0, 1000).Wait(TimeSpan.FromMinutes(1));
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/MigratorCompatibilitySpec.cs:35]: can we call `await` here?

File: src/Akka.Persistence.Linq2Db.HelperLib/Akka.Persistence.Linq2Db.HelperLib.csproj [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+        <LangVersion>10</LangVersion>
     </PropertyGroup>
 
     <ItemGroup>
       <ProjectReference Include="..\Akka.Persistence.Sql.Linq2Db\Akka.Persistence.Sql.Linq2Db.csproj" />
     </ItemGroup>
 
     <ItemGroup>
-      <PackageReference Include="FluentMigrator" Version="$(FluentMigratorVersion)" />
+      <PackageReference Include="FluentMigrator" Version="3.3.1" />
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Linq2Db.HelperLib/Akka.Persistence.Linq2Db.HelperLib.csproj]: Probably should still use the sync variable here

File: src/Akka.Persistence.Linq2Db.TagTableMigration/Options.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        => db switch
+        {
+            DatabaseType.SqlServer => "sql-server",
+            DatabaseType.Sqlite => "sqlite",
+            DatabaseType.PostgreSql => "postgresql",
+            DatabaseType.MySql => "mysql",
+            _ => throw new ArgumentException($"Unknown database type: {db}")
+        };
+
+    public static string ToHocon(this ProviderType type)
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Linq2Db.TagTableMigration/Options.cs:54]: Boilerplate for the CLI args --> Linq2Db configuration pipeline by the looks of it

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
One minor change re project structure but otherwise I think this looks good.

This is designed to provide an additional way to migrate data to the new tag table format aside from the SQL Scripts.

File: src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/MigratorCompatibilitySpec.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/MigratorCompatibilitySpec.cs:35]: Nevermind, `WaitAsync` isn't available on all of the frameworks being targeted in this test. My bad.

File: src/Akka.Persistence.Linq2Db.HelperLib/Akka.Persistence.Linq2Db.HelperLib.csproj (hunk 2 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Linq2Db.HelperLib/Akka.Persistence.Linq2Db.HelperLib.csproj]: All done

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**